lib/syscall_shim/include/uk/syscall.h

```diff
@@ -532,6 +526,70 @@ const char *uk_syscall_name_p(long nr);
  */
 long (*uk_syscall_r_fn(long nr))(void);
 
+/*
+ * Format flags for system call print functions `uk_snprsyscall()`  and
+ * `uk_vsnprsyscall()`
+ */
+/* Append a newline at the end of the generated string */
+#define UK_PRSYSCALL_FMTF_NEWLINE   0x1
+/* Apply syntax highlighting with ANSI color sequences */
+#define UK_PRSYSCALL_FMTF_ANSICOLOR 0x2
+
+/**
+ * Pretty prints a system call request and response to a given C-string buffer.
+ * The function ensures that the generated string is NULL terminated. The
+ * function truncates the output string if there is not enough space on the
+ * target buffer.
+ *
+ * @param buf
+ *  Reference to a buffer where the resulting string is stored
+ * @param len
+ *  Maximum length that can be used omn the buffer. If it is shorter than
+ *  the generated string, the string will be truncated but NULL-terminated
+ *  to fit into the buffer
+ * @param fmtf
+ *  Format flags that influence the generated string
+ * @param syscall_num
+ *  The system call number
+ * @param sysret
+ *  The return code of the system call
+ * @param ...
+ *  The system call arguments, each of it has to be passed as `long`
+ * @return
+ *  Number of characters of the generated string written to the buffer
+ *  (excluding terminating '\0')
+ */
+int uk_snprsyscall(char *buf, size_t maxlen, int fmtf, long syscall_num,
```

Comment:
Since we're using `size_t` here, maybe we should add an `#include <stddef.h>` in this header? The previous version of the PR didn't work for me without including this header, which should provide a definition for `size_t`.

Reply:
Weird that size_t included for KVM? This shouldn't be the case, the Xen and linuxu behavior is actually correct. Anyways, I'll include the header or use `__sz` instead. Thanks :+1: